src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -237,6 +240,12 @@ private void validateConfiguredSchema(Schema configuredSchema, FailureCollector
     com.google.cloud.bigquery.Schema bqSchema = getBQSchema(collector);
 
     FieldList fields = bqSchema.getFields();
+
+    if (fields.isEmpty()) {
```

Comment:
this is almost the same behavior as now, it will fail to deploy when there is a no-schema BQ table and a non-null configured schema.

This is what I suggest:
-  refactor this method and getSchema() to take the BQ schema as an argument
-  call getBQSchema() in both configurePipeline() and prepareRun()
-  in prepareRun(), if fields are empty, throw an exception